### PR TITLE
Auto Connection to chatroom and saving messages to database (basic)

### PR DESCRIPTION
**Pull request by jusasin**
### Description 
Adding more functionality to the chatroom, which includes an auto connection to the chatroom so that the user does not need to press connect anymore when accessing the chatroom. Authored by Warren, messageSave.js includes saving messages to the database. At the moment there is only one chatroom that shows all messages and does not currently separate the chatrooms into user to user base. We also can not save the messages to specific users and reload those messages to the chat page because we need protected routes in order to get the correct session id from each user. 
### Testing
Assigned to Johnson and will be addressed. 
### Possible Errors
No known possible errors besides the functionality is a bit wonky, will be worked on more to what we imagined once we can use the cookies properly to get session ids.

### CODE REVIEW

**Review by 39otsu — APPROVED** [review]
Beautiful quality of life update. This will improve user experience by a ton

File: chatroom-socket-server/server.js [hunk 1]
```diff
@@ -29,25 +34,24 @@ io.on("connection", (socket) => {
         });
     }
     socket.emit("users", users);
```
Comment (blake-gilmore) [inline chatroom-socket-server/server.js:36]: If this is building a list of users every time someone connects, we might be able to have a global users list that we would just add and remove from on each connect or disconnect

**Review by blake-gilmore — CHANGES_REQUESTED** [review]
Just needs adjustment for the test file to make sure the tests are still passing. Also will need to fix the merge conflict then I think it's ready. Not sure if there are any differences here in the chat page than the PR that we just merged from Jun, so I think that's worth looking over again.

File: small-talk/app/chat/page.js [hunk 2]
```diff
@@ -4,17 +4,17 @@ import ThemeLayout from '../components/ThemeLayout';
...
 	const [selectedUser, setSelectedUser] = useState(null);
 	const [privateMessages, setPrivateMessages] = useState([]);
 	const [inputMessage, setInputMessage] = useState("");
+	const [room, setRoom] = useState(""); 
 
 	useEffect(() => {
-		socket.auth = "random";
-		socket.connect();
-	
+		handleConnectButtonClick();
```
Comment (junxjr) [inline small-talk/app/chat/page.js:17]: This is great, we had to use the connect button before, but it is so much easier and nicer to have an automatic connection.

**Review by junxjr — APPROVED** [review]
Everything looks good. The automatic connection works. I didn't try the saving message part on my end since I didn't connect to the mongodb yet. The next step would be solving the individual rooms issue and take a look at the useEffect connection.

File: small-talk/app/chat/page.js [hunk 3]
```diff
@@ -43,15 +42,16 @@ export default function Chat(){
 			socket.off("private message");
 			socket.off("disconnect");
 		};
-	}, []); 
+	}, [privateMessages]);
```
Comment (blake-gilmore) [inline small-talk/app/chat/page.js:45]: Will this reconnect on every private message change?

**Comment by jusasin** [comment]
Currently commented out the test cases because there is more work being done and we plan to make the test cases better once the chatroom is more finalized. Lowered the test coverage as well to help with the continuation of work. Also began implementing the rooms aspect so that each user has their own room that another user can join.

**Comment by AlvaroH2001** [comment]
Merging this since it's met the approval numbers.

**Review by aiharc1 — COMMENTED** [review]
Looks good nice work with the automatic connection to the server and saving messages

**Pull request merged**